### PR TITLE
[Feature] Rebranding to publisher-interface

### PR DESCRIPTION
**Pull request by seancrowe**
This pull request provides all the rebranding for PublisherConnector to PublisherInterface.

All that needs to be done is the following:

- Reviewed by @brapoprod, keep in mind @brapoprod that this will create a new library in npm called @chili-publish/publisher-interface
  - Make sure I spelled everything correct
- One of us will then need to rename the repo to publisher-interface
- Merge the request
- publish-release-on-npm workflow should publish the new library

Let's discuss what we should do with @chili-publish/publisher-connecter on npm see:
https://github.com/chili-publish/publisher-connector/issues/11

### CODE REVIEW

**Review by brapoprod — APPROVED** [review]

**Comment by brapoprod** [comment]
@seancrowe I've done some other renamings as well, can you validate before merge?

File: src/PublisherInterface.ts [hunk 1]
```diff
@@ -546,3 +548,5 @@ export class PublisherConnector {
     }
   }
 }
+
+export {PublisherInterface as PublisherConnector};
```
Comment (pietervp) [inline src/PublisherInterface.ts:552]: What is the reasoning here? Why keep the name on this instance?

Comment (seancrowe) [reply src/PublisherInterface.ts:552]: Just for backwards compatibility as we have a few clients running this in production. Again there are two view points on this, which is reflected in https://github.com/chili-publish/publisher-interface/issues/11 as well.

- One view point is we try to keep the Publisher Interface as backwards compatible as possible, as per my suggest in https://github.com/chili-publish/publisher-interface/issues/11 
- The other view point, is deprecate and just force a change, as you suggested in https://github.com/chili-publish/publisher-interface/issues/11 

I do not think either view point is wrong, but lets get a consensus in https://github.com/chili-publish/publisher-interface/issues/11 about this as well.

Comment (seancrowe) [reply src/PublisherInterface.ts]: See commit 7, I removed the alias and related docs until a decision is made

**Pull request merged**